server/plugins/irc-events/message.ts

```diff
@@ -177,7 +177,11 @@ export default <IrcEventHandler>function (irc, network) {
 		chan.pushMessage(client, msg, !msg.self);
 
 		// Do not send notifications if the channel is muted or for messages older than 15 minutes (znc buffer for example)
-		if (!chan.muted && msg.highlight && (!data.time || data.time > Date.now() - 900000)) {
+		if (
+			!chan.muted &&
+			(msg.highlight || chan.notifyAll) &&
```

Comment:
This conditional was already somewhat large and I think by now we need to structure this a bit differently, or at the very least pull that out into its own function.

We have pretty much a similar conditional up in the socket events